### PR TITLE
[New] Elastic Defend Network Connectivity Errors

### PR DESCRIPTION
**Pull request by Samirbous**
image

### CODE REVIEW

File: rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+reference = "https://attack.mitre.org/techniques/T1562/001/"
+
+
+[rule.threat.tactic]
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
+
+[rule.threshold]
+field = ["host.id"]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml:51]: thoughts on agent.id?

```suggestion
field = ["agent.id"]
```

File: rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+value = 60
```
Comment (brokensound77) [inline rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml:52]: how often is this error thrown? is 60 in 9m the best ratio?

File: rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects an elevated frequency of Elastic Defend network connectivity errors originating from a single host in
+a brief time span. Such occurrences may arise due to a configuration adjustment or, conceivably, due to adversaries
+obstructing the agent's communication with the server in order to inhibit telemetry and alert transmission.
+"""
+from = "now-9m"
+index = ["logs-*"]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml:17]: In the description we scope this to elastic defend

```suggestion
index = ["logs-elastic_agent.endpoint_security*"]
```


but actually ... I am not sure if `endpoint_security` is standard or based on how the integration is configured. We could figure that out and scope it down or expand this to _any_ integration under `logs-*` and then add the `data_stream.dataset` to the threshold fields. Your call

**Review by brokensound77 — COMMENTED** [review]
move this to `/rules/cross-platform` to pass the unit `test_os_and_platform_in_query` failure

File: rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+name = "Elastic Defend Network Connectivity Errors"
+references = ["https://medium.com/csis-techblog/silencing-microsoft-defender-for-endpoint-using-firewall-rules-3839a8bf8d18"]
+risk_score = 73
+rule_id = "d1ffe49f-8e91-4e8d-84d2-b8dc9efbd841"
+severity = "high"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Impact", "Data Source: Elastic Defend"]
+type = "threshold"
+
+query = '''
+data_stream.dataset:elastic_agent.endpoint_security and log.level:"error" and message :"Couldn't connect to server"
```
Comment (Mikaayenson) [inline rules/windows/defense_evasion_elastic_defend_inactive_threshold.toml:29]: Is there a "Couldn't connect to agent" message we could use here?

**Pull request closed without merging**